app/src/main/java/com/github/fribourgsdp/radio/UserProfileActivity.kt

```diff
@@ -36,31 +50,44 @@ class UserProfileActivity : AppCompatActivity(), PlaylistAdapter.OnPlaylistClick
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_user_profile)
 
-        firebaseAuth = FirebaseAuth.getInstance()
-        checkUser()
 
-        user = loadUser(this)
+        instantiateViews()
+        try {
+             user = User.load(this)
+        } catch (e: java.io.FileNotFoundException) {
+            //this should never happen as a user is created and saved in the first activity
+            createDefaultUser()
+        }
+
```

Comment:
I think the User should be loaded before instantiating views because if it isn't, certain displays will fail, such as the users playlists for example.